src/Controller/CustomNodeEditController.php

```diff
@@ -4,40 +4,33 @@
 
 use Drupal\Core\Access\AccessResult;
 use Drupal\Core\Controller\ControllerBase;
-use Drupal\Core\Session\AccountInterface;
 use Drupal\Core\Entity\EntityDisplayRepositoryInterface;
 use Drupal\node\NodeInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
-use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 
+/**
+ * Class CustomNodeEditController.
+ *
+ * @package strawberryfield
+ */
 class CustomNodeEditController extends ControllerBase {
 
   /**
    * The EntityDisplayRepository service.
    *
-   * @var \Drupal\core\Entity\EntityDisplayRepository $entityDisplayRepository
-   *
+   * @var \Drupal\core\Entity\EntityDisplayRepository
    */
   protected $entityDisplayRepository;
 
   /**
-   * Node form modes active for this current user
-   *
-   * @var array
-   *
-   */
-  protected $activeNodeFormModes;
-  
-  /**
-   * Constructs a new CustomNodeEditLocalTasks
+   * Constructs a new CustomNodeEditLocalTasks.
    *
    * @param \Drupal\Core\Entity\EntityDisplayRepositoryInterface $entity_display_repository
-   *
+   *   The EntityDisplayRepository service.
    */
   public function __construct(EntityDisplayRepositoryInterface $entity_display_repository) {
     $this->entityDisplayRepository = $entity_display_repository;
-    $this->activeNodeFormModes = $this->getActiveNodeFormModes();
```

Comment:
I may fight this one but open to listen to arguments! With this change we call now ::getActiveNodeFormModes twice and each one calls again the list from storage (I know it may be cached but cached is still DB) and does the array_filter. Having it saved as protected implies we do not need to make the call again and filter again but also that in a "very very edge" case where the active ones change during (yes, I'm crazy) call to the controller, our initial assessment of what is available will be consistent with our checks/redirect?

Reply:
This is what I thought too but on my tests it is only called once as I think is statically cached by default, it would be good to double check

Reply:
Oh, that is confusing? So you say the whole controller gets all its methods statically cached? That is a bit obscure. Should I test too?

Reply:
I would appreciate if you could test, in my debug sessions, it was only being called once, as I was considering adding a static. 
Otherwise we need to use a different method such as getFormModeOptions as I don't think we have the node context on construct or at least I don't know how to get it without ugly routematches calls.

Reply:
Ok, that makes sense. Will move forward with this the way it is.